File: Priyanka-and-Toys.cpp
#include <bits/stdc++.h>

using namespace std;

string ltrim(const string &);
string rtrim(const string &);
vector<string> split(const string &);

/*
 * Complete the 'toys' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY w as parameter.
 */

int toys(vector<int> w)
{
    sort(w.begin(), w.end());
    
    int count = 0;
    int limit = INT_MIN;
    
    for (int weight : w)
    {
        if (weight > limit)
        {
            count++;
            limit = weight + 4;
        }
    }
    
    return count;
}

File: Priyanka-and-Toys.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'toys' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY w as parameter.
     */

    public static int toys(List<int> w)
    {
        w.Sort();
        
        int count = 0;
        int limit = int.MinValue;
        
        foreach (int weight in w)
        {
            if (weight > limit)
            {
                count++;
                limit = weight + 4;
            }
        }
        
        return count;
    }
}

